Patch-source: https://github.com/wejn/elixir/commit/cf15f8a848034f4e8435f813e78ad7febac7da7c

(manually tweaked offsets, because GH url returns the diff with off by two context)

diff --git a/bin/elixir b/bin/elixir
index 5a8cd9d87f..c891378475 100755
--- a/bin/elixir
+++ b/bin/elixir
@@ -234,7 +234,7 @@ set -- "$ERTS_BIN$ERL_EXEC" -pa "$SCRIPT_PATH"/../lib/*/ebin $ELIXIR_ERL_OPTIONS
 if [ -n "$RUN_ERL_PIPE" ]; then
   ESCAPED=""
   for PART in "$@"; do
-    ESCAPED="$ESCAPED $(echo "$PART" | sed 's/[^a-zA-Z0-9_\-\/]/\\&/g')"
+    ESCAPED="$ESCAPED $(echo "$PART" | sed 's@[^a-zA-Z0-9_/-]@\\&@g')"
   done
   mkdir -p "$RUN_ERL_PIPE"
   mkdir -p "$RUN_ERL_LOG"
